ethereum/core/src/test/java/org/hyperledger/besu/ethereum/core/TransactionEIP1559Test.java

```diff
@@ -14,65 +14,70 @@
-  private final Wei expectedGasPremium = Wei.of(527);
-  private final Wei expectedFeeCap = Wei.of(369);
+  private static final Supplier<SignatureAlgorithm> SIGNATURE_ALGORITHM =
+      Suppliers.memoize(SignatureAlgorithmFactory::getInstance);
 
   @After
   public void reset() {
     ExperimentalEIPs.eip1559Enabled = ExperimentalEIPs.EIP1559_ENABLED_DEFAULT_VALUE;
   }
 
   @Test
-  public void givenLegacyTransaction_assertThatRlpEncodingWorks() {
-    final Transaction legacyTransaction = Transaction.readFrom(legacyRLPInput);
-    assertThat(legacyTransaction.getType()).isEqualTo(TransactionType.FRONTIER);
-  }
-
-  @Test
-  public void givenEIP1559Transaction_assertThatRlpDecodingWorks() {
-    ExperimentalEIPs.eip1559Enabled = true;
-    final Transaction legacyTransaction = Transaction.readFrom(legacyRLPInput);
-    set(legacyTransaction, "transactionType", TransactionType.EIP1559);
-    set(legacyTransaction, "gasPrice", null);
-    set(legacyTransaction, "gasPremium", expectedGasPremium);
-    set(legacyTransaction, "feeCap", expectedFeeCap);
-    final BytesValueRLPOutput rlpOutput = new BytesValueRLPOutput();
-    legacyTransaction.writeTo(rlpOutput);
-    final Transaction eip1559Transaction =
-        Transaction.readFrom(new BytesValueRLPInput(rlpOutput.encoded(), false));
-    assertThat(legacyTransaction.getType()).isEqualTo(TransactionType.EIP1559);
-    assertThat(eip1559Transaction.getGasPremium()).hasValue(expectedGasPremium);
-    assertThat(eip1559Transaction.getFeeCap()).hasValue(expectedFeeCap);
-    assertThat(eip1559Transaction.getGasPrice()).isEqualByComparingTo(Wei.ZERO);
+  public void buildEip1559Transaction() {
+    final List<AccessListEntry> accessListEntries =
+        List.of(
+            new AccessListEntry(
+                Address.fromHexString("0x000000000000000000000000000000000000aaaa"),
+                List.of(Bytes32.ZERO)));
+    final Transaction tx =
+        Transaction.builder()
+            .chainId(new BigInteger("1559", 10))
+            .nonce(0)
+            .value(Wei.ZERO)
+            .gasLimit(30000)
+            .gasPremium(Wei.of(2))
+            .payload(Bytes.EMPTY.trimLeadingZeros())
+            .feeCap(Wei.of(new BigInteger("5000000000", 10)))
+            .gasPrice(null)
+            .to(Address.fromHexString("0x000000000000000000000000000000000000aaaa"))
+            .accessList(accessListEntries)
+            .guessType()
+            .signAndBuild(
```

Comment:
Not important but we could probably use the `BlockDataGenerator` 1559 transaction generator here